purchase_work_acceptance/models/account_invoice.py

```diff
@@ -0,0 +1,169 @@
+            'invoice_line_tax_ids': invoice_line_tax_ids.ids
+        }
+        account = invoice_line.get_invoice_line_account(
+            'in_invoice', line.product_id,
+            line.wa_id.purchase_id.fiscal_position_id, self.env.user.company_id
+        )
+        if account:
+            data['account_id'] = account.id
+        return data
+
+    def _onchange_product_id(self):
+        domain = super(AccountInvoice, self)._onchange_product_id()
+        if self.wa_id:
+            self.uom_id = self.product_id.uom_po_id
+        return domain
+
+    @api.onchange('wa_id')
+    def wa_id_change(self):
+        self.invoice_line_ids = False
+        if not self.wa_id:
+            return {}
+        if not self.partner_id:
+            self.partner_id = self.wa_id.partner_id.id
+        vendor_ref = self.wa_id.partner_ref
+        if vendor_ref and (not self.reference or (
+                vendor_ref + ", " not in self.reference and
+                not self.reference.endswith(vendor_ref))):
+            self.reference = ", ".join([self.reference, vendor_ref]) \
+                if self.reference else vendor_ref
+
+        if not self.invoice_line_ids:
+            self.currency_id = self.wa_id.currency_id
+
+        new_lines = self.env['account.invoice.line']
+        for line in self.wa_id.wa_line_ids:
+            data = self._prepare_invoice_line_from_wa_line(line)
+            new_line = new_lines.new(data)
+            new_line._set_additional_fields(self)
+            new_lines += new_line
+
+        self.invoice_line_ids += new_lines
+        self.payment_term_id = self.wa_id.purchase_id.payment_term_id
+        self.purchase_id = self.wa_id.purchase_id.id
+        self.origin = self.wa_id.purchase_id.name
+        self.env.context = dict(self.env.context, from_wa_id_change=True)
+        return {}
+
+    @api.onchange('currency_id')
+    def _onchange_currency_id(self):
+        if self.currency_id:
+            for line in self.invoice_line_ids.filtered(lambda r: r.wa_line_id):
+                date = self.date or self.date_invoice or fields.Date.today()
+                company = self.company_id
+                line.price_unit = line.wa_id.currency_id._convert(
+                    line.wa_line_id.price_unit, self.currency_id, company,
+                    date, round=False)
+
+    @api.model
+    def create(self, vals):
+        invoice = super(AccountInvoice, self).create(vals)
+        wa = invoice.invoice_line_ids.mapped('wa_line_id.wa_id')
+        if wa and not invoice.refund_invoice_id:
+            message = _(
+                "This vendor bill has been created from: %s"
+                ) % (",".join(
+                    ["<a href=# data-oe-model=work.acceptance data-oe-id=" +
+                     str(line.id) + ">" + line.name + "</a>" for line in wa]))
+            invoice.message_post(body=message)
+        return invoice
+
+    @api.multi
+    def write(self, vals):
```

Comment:
I am not very sure about this method. Do we still need it?
Note: if we still use exiting `def purchase_order_change()` to create invoice line as Odoo standard, I think we can just add onchange to `wa_id` just to trigger `purchase_order_change()` again. Almost no code we have to write.